src/Components/MessageList.jsx

```diff
@@ -0,0 +1,67 @@
+// @flow
+
+/* eslint-disable */
+import React from 'react';
+import { List } from 'semantic-ui-react';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+import { fetchMessagesForChannel } from '../Actions/index';
+import Message from './Message';
+import owl from '../images/avatars/owl.png';
+import type { Dispatch, State } from '../FlowTypes/Types';
+
+class MessageList extends React.Component {
+  props: {
+    selectedChannel: ?string,
+    messages: {},
+  }
+
+  render() {
+    const { selectedChannel } = this.props;
+    let { messages } = this.props;
+    if (!messages) {
+      if (selectedChannel) {
+        setTimeout(
```

Comment:
The render function should just be a pure function that returns React elements (or JSX). This code should go either in lifecycle methods or another utils function